cosmic-greeter: add cosmic-randr to PATH; nixos/cosmic-greeter: install cosmic-comp by default

HeitorAugustoLN opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Fixes #448617 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

HeitorAugustoLN committed:
cosmic-greeter: add missing cosmic-randr and orca dependency

The path to the orca screen reader was hardcoded to /usr/bin/orca, which prevents it from being executed in a NixOS environment.

This change adds orca as a dependency and patches the source code to use the Nix-provided executable path, allowing the greeter to launch it correctly.

Also adds cosmic-randr to PATH.

HeitorAugustoLN committed:
nixos/cosmic-greeter: install cosmic-comp, cosmic-greeter by default

JohnRTitor reviewed (approved):
I am seeing a screen now and not a blinking cursor. Can move the cursor around but still no login screen. 

This change is itself valid, but I wonder what else is left to do. 


https://github.com/user-attachments/assets/50bd6e3f-9a85-4882-ba6a-7e74ac4bf5da

HeitorAugustoLN committed:
nixos/cosmic-greeter: create `/run/cosmic-greeter` directory

Also uses the right home mode for `cosmic-greeter` user

HeitorAugustoLN committed:
nixos/cosmic-greeter: add video group to user

This change aligns the NixOS module with the permissions configured by the upstream project.

Pandapip1 commented on nixos/modules/services/display-managers/cosmic-greeter.nix:
```diff
@@ -50,14 +52,21 @@ in
       home = "/var/lib/cosmic-greeter";
-      createHome = true;
+      homeMode = "0750";
+      createHome = false;
```
Have you tested to make sure this does still work with the `/var/lib/cosmic-greeter` removed? It's not empty on my machine; the `.cache`, `.config`, and `.local` folders are there, with the cache populated, `.config` containing actual configuration, and `.local` containing... something, it seems important (https://github.com/pop-os/cosmic-comp/issues/1040)

HeitorAugustoLN commented:
@JohnRTitor Can you try removing `$XDG_STATE_HOME/cosmic-comp/outputs.ron` to renegerate it too se if it works?

JohnRTitor commented:
Still it doesn't work.

HeitorAugustoLN commented on nixos/modules/services/display-managers/cosmic-greeter.nix:
```diff
@@ -50,14 +52,21 @@ in
+    systemd.tmpfiles.rules = [
+      "d /var/lib/cosmic-greeter 0750 cosmic-greeter cosmic-greeter -"
+      "d /run/cosmic-greeter 0755 cosmic-greeter cosmic-greeter -"
+    ];
```
I wanted to centralize it on tmpfiles, but I've changed my mind

HeitorAugustoLN commented:
I don't think there is anything else to do on our side

JohnRTitor reviewed (approved):
Some of the commits here can be squashed together. And instead of delaying this PR any further, I will go ahead and do it myself. I hope you don't mind.
